src/grass_gis_helpers/validation.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:       lib with validation related helper functions for GRASS GIS
+#
+# AUTHOR(S):    Lina Krisztian
+#
+# PURPOSE:      lib with validation related helper functions for GRASS GIS
+#
+# COPYRIGHT:	(C) 2023 by mundialis and the GRASS Development Team
+#
+# 		This program is free software under the GNU General Public
+# 		License (>=v2). Read the file COPYING that comes with GRASS
+# 		for details.
+#
+#############################################################################
+
+import subprocess
+
+import grass.script as grass
+
+
+def check_valid_rasterdata(input):
+    """Check if input is not broken"""
+    gdalinfo_cmd = ["gdalinfo", "-mm", input]
+    p_gdalinfo = subprocess.Popen(
+        gdalinfo_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    gdalinfo_err = p_gdalinfo.communicate()[1]
+    if gdalinfo_err.decode("utf-8") != "":
```

Comment:
This test is maybe a bit too strict. Sometimes there are mostly harmless error messages written by gdalinfo to stderr, but the data can be read correctly anyway. If gdalinfo fails with a returncode != 0, then the rasterdata are definitively broken.

Reply:
Maybe it is possible to have two cases? (I think in general a developer most likely only uses this method in certain cases when there is already the suspicion that the data might be corrupt)